### DISCUSSION TITLE
Is it recommended to use pocketbase with nextjs?

### DISCUSSION THREAD

**Discussion by thal-sh** [pocketbase/pocketbase#4973] [Q&A]
I do realize that there are workarounds for server components and such, but at the end of the day is it worth it to use with nextjs or it is better to use another framework?

**Comment by lumosminima** [comment]
Pocketbase is a "backend" so it doesn't matter what front-end you use. In next.js or sveltekit etc you can either create a client only SPA which will talk to pocketbase via the pocketbase sdk or you can use a [BFF architecture](https://learn.microsoft.com/en-us/azure/architecture/patterns/backends-for-frontends) which will use SSR and the pocketbase sdk inside serverless functions. Personally I have used both and can recommend sveltekit for headache free development and if you already know svelte but if react is needed I guess Remix is the alternative to next.js.

**Reply by thal-sh** [reply]
ye I do realize that it's a backend and it doesn't matter what I pick, but the creator himself said that nexjts is a hassle and not even worth it, and tbh most of the time I got pocketbase errors when I've used nexjts lol, I guess I'll use supabase for my nextjs apps and for client side apps pocketbase would be really good.
